src/fluree/db/migrations/nameservice_v1.cljc

```diff
@@ -0,0 +1,160 @@
+  
+  WHAT IT DOES:
+  - Detects old nameservice files stored at root level (e.g., ledger-name.json)
+  - Extracts essential metadata from full commit JSON-LD records
+  - Creates new minimal records with only: commit address, t value, index address
+  - Stores them in the new ns@v1/ directory with branch-aware naming (ledger-name@branch.json)
+  - Cleans up old files after successful migration
+  
+  WHEN IT RUNS:
+  - Automatically at file storage initialization if old format is detected
+  - Only runs once when ledger directories exist but no ns@v1/ directory is present
+  
+  SAFETY:
+  - Read-only detection phase before any modifications
+  - Sequential processing with proper error handling
+  - Old files only deleted after successful migration
+  - Logging for monitoring and debugging"
+  (:require #?(:clj [clojure.java.io :as io])
+            [clojure.string :as str]
+            [fluree.db.nameservice.storage :as ns-storage]
+            [fluree.db.storage :as storage]
+            [fluree.db.util.async :refer [<? go-try]]
+            [fluree.db.util.filesystem :as fs]
+            [fluree.db.util.json :as json]
+            [fluree.db.util.log :as log]))
+
+(defn needs-migration?
+  "Check if nameservice migration is needed by detecting old format files without ns@v1 directory"
+  [file-store]
+  (go-try
+    (let [root-path   (:root file-store)
+          root-files  (<? (fs/list-files root-path))
+          has-ledgers? (some #(and (not= % "ns@v1")
+                                   (not (str/includes? % ".")))
+                             root-files)
+          has-ns-v1?   (some #(= % "ns@v1") root-files)]
+      (and has-ledgers? (not has-ns-v1?)))))
+
+(defn find-json-files-recursively
+  "Find all JSON files in directory tree, excluding certain paths"
+  [root-path]
+  #?(:clj
+     (let [root-file (io/file root-path)]
+       (->> (file-seq root-file)
+            (filter #(.isFile ^java.io.File %))
+            (map #(.getPath ^java.io.File %))
+            (filter #(and (str/ends-with? % ".json")
+                          (not (str/includes? % "/ns@v1/"))
+                          (not (str/includes? % "/commit/"))
+                          (not (str/includes? % "/index/"))
+                          (not (str/includes? % "/txn/"))))))
+     :cljs
+     (throw (ex-info "Migration not supported in ClojureScript" {:path root-path}))))
+
+(defn find-old-nameservice-files
+  "Find old nameservice files in ledger directories - handles both root level and nested paths"
+  [file-store]
+  (go-try
+    (let [root-path   (:root file-store)
+          ;; Find all .json files recursively, excluding ns@v1 and commit directories
+          all-json-files (find-json-files-recursively root-path)
+          ;; Filter to only nameservice files (those that match ledger naming pattern)
+          ;; clj-kondo false positive - ledger-path is used in the clj branch  
+          ns-files #_{:clj-kondo/ignore [:unused-binding]}
```

Comment:
I think this would be better in the clj-kondo configuration files to avoid adding noise to the code.